Lib/test/test_capi/test_misc.py

```diff
@@ -2263,6 +2263,44 @@ def test_module_state_shared_in_global(self):
         subinterp_attr_id = os.read(r, 100)
         self.assertEqual(main_attr_id, subinterp_attr_id)
 
+    def test_datetime_capi_type_check(self):
```

Comment:
All our other tests of the datetime C-API are methods of `CapiTest` in Lib/test/datetimetester.py.  Is there any reason this test should be different?